### ISSUE TITLE
[BUG] C# Dev Kit - Test Explorer does not honor `RunSettingsFilePath` project file property

### ISSUE THREAD

**Issue by ChristoWolf** [microsoft/vscode-dotnettools#835]
### Describe the Issue

Hi!

Since [this feature addition](https://github.com/microsoft/vstest/issues/605), it is possible to specify .runsettings files for automatic use during test runs, see also [this](https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022#set-a-build-property).

However, this property seems to not be honored when running tests using VS Code's TESTING view, which seems to trigger tests using the C# Dev Kit - Test Explorer (according to its log output).

### Steps To Reproduce

1. Define 
   ```xml
    <PropertyGroup>
      <RunSettingsFilePath>$(MSBuildProjectDirectory)\..\..\.runsettings</RunSettingsFilePath>
    </PropertyGroup>
   ```
   (or whatever your path to a valid .runsettings file is) in some test project.
2. Run any test(s) contained in the project by running them from VS Code's TESTING view.
3. Check if the run settings where applied correctly.

### Expected Behavior

The run settings are applied the same way as when running `dotnet test` directly on the project.
E.g. when it specifies a global `ResultsDirectory`
```xml
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <RunConfiguration>
    <ResultsDirectory>.\TestResults</ResultsDirectory>
  </RunConfiguration>
</RunSettings>
```
, test results should be written to that folder which should then automatically exist in the .runsettings file's directory.

### Environment Information

- OS: Microsoft Windows version 22H2 (OS Build 19045.3693)
- VS Code version: 1.85.1
- Extension version: 1.1.16
- ,NET SDK version: 8.0.100

**Comment by ChristoWolf** [comment]
Duplicate of #156.

**Comment by leo-labs** [comment]
Is this really a duplicate of #156? #156 is about setting the `.runsetting` file via vs code configuration `dotnet.unitTests.runSettingsPath` but this is about the C# Dev Kit ignoring the `<RunSettingsFilePath>` property in the `.csproj` file. 

Please reopen :)

**Comment by ChristoWolf** [comment]
Yeah I am honestly not sure.

After #156, the `.runsettings` file is definitely honored when pointing `dotnet.unitTests.runSettingsPath` to it .
But I am pretty sure that that is not the case with just `RunSettingsFilePath` being configured.

Also, when the `.runsettings` file is honored, its root folder is definitely not considered as its reference path as required by [this property](https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022#:~:text=of%20testhosts%20started.-,ResultsDirectory,The%20path%20is%20relative%20to%20the%20directory%20that%20contains%20.runsettings%20file.,-TargetFrameworkVersion).

**Comment by Meligy** [comment]
@@peterwald not entirely sure this is an enhancement not a bug. You'd expect a test runner to use a csproj, and the property is in the csproj.

**Comment by iongion** [comment]
This did it for me `dotnet.unitTests.runSettingsPath`, using like this `"dotnet.unitTests.runSettingsPath": "${workspaceFolder}/Api.Tests/Tests.runsettings" in `settings.json`

Works for everything, SpecFlow, MS Tests and xUnit, example `Tests.runsettings`

```xml
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <RunConfiguration>
    <EnvironmentVariables>
      <ENVIRONMENT>Testing</ENVIRONMENT>
      <ASPNETCORE_ENVIRONMENT>Testing</ASPNETCORE_ENVIRONMENT>
    </EnvironmentVariables>
    <TreatNoTestsAsError>false</TreatNoTestsAsError>
  </RunConfiguration>
</RunSettings>
```

And in the `Api.Tests.csproj`

```
<PropertyGroup>
     <RunSettingsFilePath>$(MSBuildProjectDirectory)\Tests.runsettings</RunSettingsFilePath>
</PropertyGroup>
```

**Comment by gmkado** [comment]
I am also running into the issue where the `ResultsDirectory` is not relative to the .runsettings file, but rather the workspace root.

This file is in `<workspace>/tests/.runsettings` (set via `dotnet.unitTests.runSettingsPath`)
```xml
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <RunConfiguration>
    <ResultsDirectory>./TestResults</ResultsDirectory>
  </RunConfiguration>
  <LoggerRunSettings>
    <Loggers>
      <Logger friendlyName="trx" enabled="True">
        <Configuration>
          <LogFileName>TestResults.trx</LogFileName>
        </Configuration>
      </Logger>
    </Loggers>
  </LoggerRunSettings>
</RunSettings>
```
I would expect the results to go to `<workspace>/tests/TestResults/TestResults.trx` but they end up in `<workspace>/TestResults/TestResults.trx`
